src/Moq/Language/Flow/SetupSequencePhrase.cs

```diff
@@ -71,12 +71,34 @@ public ISetupSequentialResult<TResult> Returns(Func<TResult> valueFunction)
 			if (valueFunction is TResult)
 			{
 				return this.Returns((TResult)(object)valueFunction);
-			}
+			}	
 
 			this.setup.AddReturns(() => valueFunction());
 			return this;
 		}
 
+		public ISetupSequentialResult<TResult> ReturnsAsync(Func<TResult> valueFunction)
+		{
+			Guard.NotNull(valueFunction, nameof(valueFunction));
+
+			// If `valueFunction` is `TResult`, that is someone is setting up the return value of a method
+			// that returns a `TResult`, then we have arrived here because C# picked the wrong overload:
+			// We don't want to invoke the passed delegate to get a return value; the passed delegate
+			// already is the return value.
+			if (valueFunction is TResult)
+			{
+				return this.Returns((TResult)(object)valueFunction);
+			}
```

Comment:
Have you verified (before copy-and-pasting) whether this check still makes sense for your new method?